app/services/island_service.rb

```diff
@@ -0,0 +1,57 @@
+class IslandService
+
+  ITERATIONS = 8
+
+  def initialize(position, size)
+    @position = position
+    @size = size
+
+  end
+
+  def call
+    island = create_polygon
+  end
+
+  def generate_midpoint(one_edge)
+    pt1 = one_edge.first
+    pt2 = one_edge.last
+    mx = (pt1.first + pt2.first) / 2
+    my = (pt1.last + pt2.last) / 2
+    vx = -(pt1.last - pt2.last)
+    vy =  (pt1.first - pt2.first)
+    d = [-1, 1].sample * Random.rand/2
+    [(mx + (d * vx)),(my + (d * vy))]
```

Comment:
'd' could be a  parameter instead to get rid of the random when testing 

also maybe naming could be nicer here. pt1, pt2, mx,my,vx,vy and d are no speaking names  

also the function name itself could be less describing. Try function names that rather describe what the result is, rather than adding a verb. maybe 'midpoint' or 'displaced_midpoint or 'altered_midpoint' or 'orthogonal_midpoint' (dunno what the proper mathematical term is but that would be great)